lib/datadog/opentelemetry/sdk/trace/span.rb

```diff
@@ -15,6 +15,20 @@ def set_attribute(key, value)
           res
         end
 
+        def record_exception(exception, attributes: nil)
```

Comment:
Can the arguments of this method be documented? It is unclear why data in `attributes` is overriding the attributes of `exception`, i.e. what the relationship should be between these arguments.

Reply:
This convention comes from how opentelemetry manages Exeception Events. The attribute map can be used to override the error message, traceback, and error type associated with an exception object: https://github.com/open-telemetry/opentelemetry-ruby/blob/35abeac8653359aebd0231c94a23587ad4ee372e/sdk/lib/opentelemetry/sdk/trace/span.rb#L170. I just copied this logic to map to the Span's error fields.

Reply:
I think it's worth adding to do documentation of this method that it has the same interface as:
https://github.com/open-telemetry/opentelemetry-ruby/blob/406cabe54f8fb4937d0f6b73fe830a51c04fdb92/api/lib/opentelemetry/trace/span.rb#L126-L136